src/ansys_sphinx_theme/extension/autoapi.py

```diff
@@ -0,0 +1,151 @@
+"""AutoAPI extension for the ANSYS Sphinx theme."""
+
+# Copyright (C) 2021 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import pathlib
+from typing import Any, Dict
+
+from sphinx.application import Sphinx
+
+from ansys_sphinx_theme import __version__
+
+THIS_PARENT_PATH = pathlib.Path(__file__).parent.absolute()
+
+
+AUTOAPI_TEMPLATES_PATH = (
+    THIS_PARENT_PATH.parent.resolve() / "theme" / "ansys_sphinx_theme" / "_templates" / "autoapi"
+)
+
+
+def get_autoapi_templates_dir_relative_path(path: pathlib.Path) -> str:
+    """Return a string representing the relative path for autoapi templates.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        Path to the desired file.
+
+    Returns
+    -------
+    str
+        A string rerpesenting the relative path to the autoapi templates.
+
+    """
+    print(AUTOAPI_TEMPLATES_PATH.absolute())
+    return os.path.relpath(str(AUTOAPI_TEMPLATES_PATH.absolute()), start=str(path.absolute()))
+
+
+def replace_html_tag(app, exception):
```

Comment:
We can split #362. This pull-request can focus on shipping autoapi as an extension on our side and #362 can focus on nerd-fonts.

Reply:
Sure adapting that.